Protocol.md

```diff
@@ -23,188 +45,183 @@ The following figure gives a brief overview of the protocol:
+  - the funder's amount for the funding transaction is not sufficient for their full [fee payment](Transactions.md#fee-payment).
 
-The following addresses:
+### The `accept_dlc` Message
 
-1. Funding Change Address
-   - The address to use to send the change if the sum of A's inputs is greater than the fund amount plus the fees.
-2. Final Address
-   - The address to which funds will be sent in [unilateral contract execution](Transactions.md#CETOutputs), [refund](Transactions.md#refund-transaction) and [mutual closing](Transactions.md#mutual-closing-transaction) transactions.
+This message contains information about a node and indicates its
+acceptance of the new DLC, as well as its CET and refund transaction
+signatures. This is the second step toward creating the funding transaction
+and closing transactions.
 
-#### B's inputs
-The set of UTXOs to be used as input to the fund transaction.
+1. type: ??? (`accept_dlc`)
+2. data:
+   * [`32*byte`:`temporary_contract_id`]
+   * [`u64`:`total_collateral_satoshis`]
+   * [`point`:`funding_pubkey`]
+   * [`spk`:`payout_spk`]
+   * [`u16`:`num_funding_inputs`]
+   * [`num_funding_inputs*funding_input`:`funding_inputs`]
+   * [`spk`:`change_spk`]
+   * [`cet_signatures`:`cet_signatures`]
+   * [`signature`:`refund_signature`]
 
-#### CET signatures
-A set of signatures from B, one for each CET input.
+#### Requirements
 
-#### Refund signature
-A signature from B for the refund transaction input.
+The `temporary_contract_id` MUST be the SHA256 hash of the `offer_dlc` message.
 
----
+The sender MUST:
 
-### Sign
+  - set `total_collateral_satoshis` sufficiently large so that the sum of both parties' total collaterals is at least as large as the largest payout in the `offer_dlc`'s `contract_info`.
+  - set `cet_signatures` to valid adaptor signatures, using its `funding_pubkey` for each CET, as defined in the [transaction specification](Transactions.md#contract-execution-transaction) and using signature public keys computed using the `offer_dlc`'s `contract_info` and `oracle_info` as adaptor points.
+  - include an adaptor signature in `cet_signatures` for every event specified in the `offer_dlc`'s `contract_info`.
+  - set `refund_signature` to the valid signature, using its `funding_pubkey` for the refund transaction, as defined in the [transaction specification](Transactions.md#refund-transaction).
 
-After receiving and [validating](#sign-validation) the content of the `offer` message, A replies with a `sign` message containing:
-1. Fund transaction signatures
-1. CET signatures
-1. Refund signature
+The receiver:
 
-#### Fund transaction signatures
-A set of signatures from A, one (in case of P2WPKH) or several (in case of multisig P2WSH) for each of A's input to the fund transaction.
+  - if `total_collateral_satoshis` is not large enough:
+    - MAY reject the contract.
+  - if `cet_signatures` or `refund_signature` fail validation:
+    - MUST reject the contract.
```

Comment:
Let's not do what LN did, should specify that they must verify the funding inputs have enough to pay the `total_collateral` + fees